network_simulator/link.py

```diff
@@ -95,3 +164,118 @@ def __eq__(self, other):
 
     def __ne__(self, other):
         return not self.__eq__(other)
+
+    def get_link_cost(self):
+        """
+        Returns link cost. For now, just returns static propagation delay.
+        :return: float of link delay time in seconds
+        """
+
+        # TODO consider average wait time for last N packets
+        return self.static_delay_sec
+
+    def push(self, packet, global_clock_sec):
+        """
+        Pushes packet onto own LinkBuffer
+
+        :param Packet packet: Packet to push
+        :param global_clock_sec: store when packet pushed
+        :return: True if pushed, False if not
+        """
+        return self.link_buffer.push(packet, global_clock_sec)
+
+
+
+class LinkSendEvent(Event):
+    """
+    Event representing a packet arriving at the Link
+    """
+
+    def __init__(self, link):
+        """
+        :ivar link: link that must send its next packet
+        :ivar buff_elem: temporary variable containing LinkBufferElement to send
+        """
+        self.link = link
+
+    def run(self, main_event_loop, statistics):
+        """
+        runs by popping off top packet
+        :param MainEventLoop main_event_loop: event loop for global timing
+        :param Statistics statistics: the Statistics to update
+        """
+
+        # sanity check
+        if(self.link.queue.empty()):
+            logger.warning("LinkSendEvent while Link empty!")
+            return
+
+        # now pop
+        self.buffer_elem = self.link.pop()
+
+        # update statistics only if not RouterPacket
+        if(not isinstance(self.buffer_elem.packet, RouterPacket)):
```

Comment:
nit: no parentheses around what's inside the if in Python. Just do "if not isinstance... :"